Fun.Build/ConditionsBuilder.fs

```diff
@@ -32,6 +32,17 @@ module Internal =
                 false
             | Mode.Execution -> isTrue
 
+        member ctx.WhenStage(stage: StageContext) =
+            match ctx.GetMode() with
+            | Mode.Execution ->
+                let result, exns = stage.Run(StageIndex.WhenStage, System.Threading.CancellationToken.None)
```

Comment:
I think we should change the stage ParentContext to ctx before run. So we can get the EnvVars, WorkingDir etc when run commands.

Reply:
It is also better to add one more test for checking if the parent context is assigned successful.